exp/addons/internal/controllers/clusterresourceset_controller_test.go

```diff
@@ -970,13 +981,22 @@ metadata:
 		testCluster.SetLabels(labels)
 		g.Expect(env.Update(ctx, testCluster)).To(Succeed())
 
-		// ClusterResourceSetBinding for the Cluster is not created because the Kubernetes API Server Service doesn't exist.
+		// Resources are not applied because the Kubernetes API Server Service doesn't exist.
 		clusterResourceSetBindingKey := client.ObjectKey{Namespace: testCluster.Namespace, Name: testCluster.Name}
 		g.Consistently(func() bool {
 			binding := &addonsv1.ClusterResourceSetBinding{}
 
-			err := env.Get(ctx, clusterResourceSetBindingKey, binding)
-			return apierrors.IsNotFound(err)
+			if err := env.Get(ctx, clusterResourceSetBindingKey, binding); err != nil {
+				// either the binding is not there
+				return true
```

Comment:
I think we should return false here.

The goal should be to verify the CRS was reconciled, but the resources were intentionally not applied.

With the current implementation it could be just racy if the CRS controller doesn't create the CRSB quick enough